### PR TITLE
feat(connect): automatically update outdated device language

### PR DESCRIPTION
**Pull request by mroz22**
based on #11300

- moves logic from `packages/connect/src/api/changeLanguage.ts` to `packages/connect/src/device/Device.ts` without any changes
- adds automatic language update if `!this.features.language_version_matches` whenever `device.getFeatures` got called. This should get triggered only once per firmware version. [Related discussion here](https://satoshilabs.slack.com/archives/C02V2PSDNA2/p1709672892741739)

There is still one open question:
- user updates from `2.6.4` to `2.7.0`
- this means automatic language upadte during fw update can't take place here (not supported by 2.6.4 yet)
- this means user is on `2.7.0` and his device still speaks english. Now user can change language manually. Next fw update should also update his language using this new feature added in this PR.
- is that enough? Or do we already want to push language update when upgrading to 2.7.0? I guess we do want because without this, language would set only after second firmware update after device purchase. 🤔 

cc @komret @Hannsek

### CODE REVIEW

**Comment by Hannsek** [comment]
> is that enough? Or do we already want to push language update when upgrading to 2.7.0? I guess we do want because without this, language would set only after second firmware update after device purchase. 🤔

I think it is enough. After the device purchase users now install 2.7.0 and the language will be uploaded into the device right after the fw installation. For those who updates **to** 2.7.0 it would be really weird if the language of the device would change automatically.

So no need to push language update when upgrading to 2.7.0. IMO it is undesirable. cc @Hermez-cz

**Comment by mroz22** [comment]
ok. better for me. 

just one last thing to clarify - without any further changes, automatic change of language from english to any other changes will NEVER happen. user first needs to change language manually in settings and only after that automatic language updates will apply.

**Comment by Hannsek** [comment]
yes, automatic **change** should never happen.

Either users install firmware and language on the device will be set to the same language as Suite right away, or users change the language of the device in settings.


One thing we should consider. When users change the language of the Suite in settings and we support the same language for Trezor device, we should offer the language change on Trezor to the user (using some modal 😈, or something… this is up to @Hermez-cz)

**Review by komret — APPROVED** [review]

File: packages/connect/src/device/Device.ts [hunk 1]
```diff
@@ -527,6 +528,76 @@ export class Device extends TypedEmitter<DeviceEvents> {
...
+    }: { language?: undefined; binary: ArrayBuffer } | { language: string; binary?: undefined }) {
+        if (language === 'en-US') {
+            return this._uploadTranslationData(null);
+        }
+
+        if (binary) {
+            return this._uploadTranslationData(binary);
+        }
+
+        const downloadedBinary = await getLanguage({
```
Comment (mroz22) [inline packages/connect/src/device/Device.ts:559]: silent error when offline

**Pull request merged**